dvc/api.py

```diff
@@ -120,94 +127,134 @@ def _make_repo(repo_url=None, rev=None):
         yield repo
 
 
-def summon(name, repo=None, rev=None, summon_file="dvcsummon.yaml", args=None):
+def summon(name, repo=None, rev=None, summon_file=DEF_SUMMON, args=None):
     """Instantiate an object described in the `summon_file`."""
-    with prepare_summon(
-        name, repo=repo, rev=rev, summon_file=summon_file
-    ) as desc:
+    with SummonDesc.prepare_summon(repo, rev, summon_file) as desc:
+        dobj = desc.get_dobject(name)
         try:
-            summon_dict = SUMMON_PYTHON_SCHEMA(desc.obj["summon"])
+            summon_dict = SUMMON_PYTHON_SCHEMA(dobj["summon"])
         except Invalid as exc:
             raise SummonError(str(exc)) from exc
 
+        desc.pull(dobj)
         _args = {**summon_dict.get("args", {}), **(args or {})}
         return _invoke_method(summon_dict["call"], _args, desc.repo.root_dir)
 
 
-@contextmanager
-def prepare_summon(name, repo=None, rev=None, summon_file="dvcsummon.yaml"):
-    """Does a couple of things every summon needs as a prerequisite:
-    clones the repo, parses the summon file and pulls the deps.
+class SummonDesc(object):
```

Comment:
So it's now `SummonFile`. Why do you want to separate it anyway? Will we use single file to work with several objects?

Reply:
Changed to `SummonFile`.

Sure, in the resnet example we had many resnet models in a single file. `pretrained-models.pytorch` project contains 50+ models. We probably don't what to have 50 summon files (the code is localized in few files only for all the models).